catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/MetacardComparator.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.impl;
+
+import com.google.api.client.repackaged.com.google.common.base.Strings;
+
+import ddf.catalog.data.Metacard;
+
+public class MetacardComparator {
+
+    private MetacardComparator() {
+    }
+
+    public static boolean isSame(Metacard cachedMetacard, Metacard latestMetacard) {
```

Comment:
Let's rename this to `compareProducts()` if that's OK with everyone. Also, the below code can be greatly simplified: 
```
public static boolean compareProducts(Metacard reference, Metacard suspect) {
        return !(Strings.isNullOrEmpty(reference.getAttribute(Metacard.CHECKSUM)
                .toString()) || Strings.isNullOrEmpty(suspect.getAttribute(Metacard.CHECKSUM)
                .toString())) && reference.getAttribute(Metacard.CHECKSUM)
                .toString()
                .equals(suspect.getAttribute(Metacard.CHECKSUM)
                        .toString()) && reference.getModifiedDate()
                .equals(suspect.getModifiedDate()) && reference.getResourceURI()
                .equals(suspect.getResourceURI()) && reference.getContentTypeName()
                .equals(suspect.getContentTypeName()) && reference.getResourceSize()
                .equals(suspect.getResourceSize()) && reference.getCreatedDate()
                .equals(suspect.getCreatedDate()) && reference.getExpirationDate()
                .equals(suspect.getExpirationDate()) && reference.getLocation()
                .equals(suspect.getLocation()) && reference.getEffectiveDate()
                .equals(suspect.getEffectiveDate()) && reference.getId()
                .equals(suspect.getId());

    }
```